torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -272,7 +272,7 @@ class NCCLComm {
     // Add lock in this destructor, as aborted_ needs to be read after memory
     // barrier here.
     std::unique_lock<std::mutex> lock(mutex_);
-    if (ncclComm_ && !aborted_) {
+    if (ncclComm_ && initialized_ && !aborted_) {
```

Comment:
this patch does not make other calls to `ncclCommAbort` (e.g. from watchdog) safe.

should we address that?

Reply:
That's a good point, all those calls should be gated by initialized_